material/plugins/social/plugin.py

```diff
@@ -49,13 +49,24 @@
 from shutil import copyfile
 from tempfile import TemporaryFile
 from zipfile import ZipFile
+
+from .config import SocialConfig
+
 try:
-    from cairosvg import svg2png
     from PIL import Image, ImageDraw, ImageFont
-except ImportError:
-    pass
+except ImportError as e:
```

Comment:
So I'm not sure I'm for using this import error concatenation pattern, but I'm short on time now thinking of a better approach, so I'm happy to test it out. Is that a common practice?

Reply:
> Is that a common practice?

Converting the error to string is a valid practice, but I don't know if it's common. The idea with gathering the errors to print them out later came from MkDocs <-> Plugin relationship. Errors that are local to the plugin should be properly moved upstream to the event flow and I wanted to be more specific about the message that says "dependencies not installed", so I needed to store the errors from the config load to report them later on.
If you feel it's weird I can just revert it :v:

Reply:
Let's just keep it like that for now and revisit it later, all good! I just wanted to ask.

Reply:
Funnily enough, when researching things for the troubleshooting guide I found out that the cairocffi (dependency of cairosvg) also gathers the exceptions and the result we see are a mix of formatted strings and errors joined together 😅 

So the commonality increased :v:
https://github.com/Kozea/cairocffi/blob/f1984d644bbc462ef0ec33b97782cf05733d7b53/cairocffi/__init__.py#L39-L40